code/nebula/neb.cpp

```diff
@@ -811,6 +871,17 @@ void upkeep_poofs()
 		}
 	}
 
+	// cull unused poof types
+	if (!Neb2_poofs.empty()) {
+		for (size_t i = 0; i < Neb2_poofs.size();) {
+			if (!poof_is_used(Neb2_poofs[i].poof_info_index)) {
+				Neb2_poofs[i] = Neb2_poofs.back();
+				Neb2_poofs.pop_back();
+			} else // if we needed to cull we should not advance because we just moved a new poof into this spot
+				i++;
+		}
+	}
+
```

Comment:
This should be moved into the above logic ( if too far || !is_used(..) cull it ). There can be an indefinite number of poofs so this is a potential performance concern. I *think* doing it that way is faster, but more importantly I think it's a bit clearer to have a single pass for culling.